torch/csrc/jit/passes/concat_opt.cpp

```diff
@@ -0,0 +1,480 @@
+#include <torch/csrc/jit/passes/concat_opt.h>
+
+#include <algorithm>
+#include <unordered_set>
+#include <vector>
+
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/ir/named_value.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/dead_code_elimination.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+void removeCatNodeFromGraph(Node* n) {
+  TORCH_INTERNAL_ASSERT(n->kind() == aten::cat);
+  auto inp_list = n->input(0);
+  GRAPH_UPDATE("Deleting\n", *n);
+  n->destroy();
+  if (!inp_list->hasUses()) {
+    GRAPH_UPDATE("Deleting\n", *inp_list->node());
+    inp_list->node()->destroy();
+  }
+}
+
+class ConcatCseEliminator {
```

Comment:
Nit: "CseEliminator" is a tautology I think

Reply:
Right, will update the name.

Reply:
Updated.